models/mpas_atm/model_mod.f90

```diff
@@ -661,29 +661,30 @@ subroutine static_init_model()
 ! fill in the grid values
 call get_grid(ncid)
 
-! vertical faces are in the input file.  compute vertical center locations here.
-do kloc=1, nCells
-   do iloc=1, nVertLevels
-      zGridCenter(iloc,kloc) = (zGridFace(iloc,kloc) + zGridFace(iloc+1,kloc))*0.5_r8
+! zgrid is defined at full levels [nVertLevelsP1], with the first level at the surface.
+! Compute zgrid at half levels for analysis fields (ex. theta, rho, qv, and horizontal wind).
+do iloc=1, nCells
+   do kloc=1, nVertLevels
+      zGridCenter(kloc,iloc) = (zGridFull(kloc,iloc) + zGridFull(kloc+1,iloc))*0.5_r8
    enddo
 enddo
 
 if(data_on_edges) then
    ! FIXME: This code is supposed to check whether an edge has 2 neighbours or 1 neighbour and then
    !        compute the height accordingly.  HOWEVER, the array cellsOnEdge does not change with
    !        depth, but it should as an edge may have 2 neighbour cells at the top but not at depth.
-   do kloc=1, nEdges
-      do iloc=1, nVertLevels
-         cel1 = cellsOnEdge(1,kloc)
-         cel2 = cellsOnEdge(2,kloc)
+   do iloc=1, nEdges
+      do kloc=1, nVertLevels
+         cel1 = cellsOnEdge(1,iloc)
+         cel2 = cellsOnEdge(2,iloc)
          if (cel1>0 .and. cel2>0) then
-            zGridEdge(iloc,kloc) = (zGridCenter(iloc,cel1) + zGridCenter(iloc,cel2))*0.5_r8
+            zGridEdge(kloc,iloc) = (zGridCenter(kloc,cel1) + zGridCenter(kloc,cel2))*0.5_r8
          else if (cel1>0) then
-            zGridEdge(iloc,kloc) = zGridCenter(iloc,cel1)
+            zGridEdge(kloc,iloc) = zGridCenter(kloc,cel1)
          else if (cel2>0) then
-            zGridEdge(iloc,kloc) = zGridCenter(iloc,cel2)
+            zGridEdge(kloc,iloc) = zGridCenter(kloc,cel2)
          else  !this is bad...
-            write(string1,*)'Edge ',kloc,' at vertlevel ',iloc,' has no neighbouring cells!'
+            write(string1,*)'Edge ',iloc,' at vertlevel ',kloc,' has no neighbouring cells!'
```

Comment:
what's the goal of renaming (swapping) the loop variables for these two do loops? 
The functionality is the same.

Reply:
Yes, the model indices are (k,j,i), from the wrf days. So we tend to think of k as vertical index. Even within this code, we have "vloc" at the same first index, so I think it's more intuitive to go with "kloc" there.